### PR TITLE
Tremblp/hydra 1235/isolate select point instancing

### PR DESCRIPTION
**Pull request by ppt-adsk**
Point instancing isolate select support. Nested point instancing isolate select support deferred, as there is no clear immediate use case for it: selecting a nested instance does not mean you can change it.

Code is still in rough shape, with a lot of duplication and conditionals, to be cleaned up in a later pull request.

### CODE REVIEW

**Comment by ppt-adsk** [comment]
FYI @debloip-adsk @lilike-adsk.

File: lib/flowViewport/sceneIndex/fvpIsolateSelectSceneIndex.cpp [hunk 1]
```diff
@@ -454,6 +487,33 @@ IsolateSelectSceneIndex::_CollectInstancers(
     return instancers;
 }
 
+bool IsolateSelectSceneIndex::_IsPointInstancer(
+    const PXR_NS::SdfPath& primPath
+) const
+{
+    auto prim = GetInputSceneIndex()->GetPrim(primPath);
```
Comment (lanierd-adsk) [inline lib/flowViewport/sceneIndex/fvpIsolateSelectSceneIndex.cpp]: Is there a reason for not using the FVP prim utils IsPointInstancer function ?

File: lib/flowViewport/sceneIndex/fvpIsolateSelectSceneIndex.cpp [hunk 2]
```diff
@@ -489,6 +576,7 @@ IsolateSelectSceneIndex::_CreateInstancerMasks(
         std::size_t i=0;
         for (const auto& instanceLocation : instanceLocations) {
             const bool included = isolateSelection->HasAncestorOrDescendantInclusive(instanceLocation);
+
```
Comment (lanierd-adsk) [inline lib/flowViewport/sceneIndex/fvpIsolateSelectSceneIndex.cpp]: I would remove that extra line unless there is a desire to do a separation ?

File: test/lib/mayaUsd/render/mayaToHydra/cpp/testUsdPointInstancingIsolateSelect.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import fixturesUtils
+import mtohUtils
+import mayaUsd
+import mayaUsd_createStageWithNewLayer
```
Comment (lanierd-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/cpp/testUsdPointInstancingIsolateSelect.py]: Is this used in the file ?

**Review by lanierd-adsk — COMMENTED** [review]
A few comments, but looks good overall. Thanks for that work.

File: test/lib/mayaUsd/render/mayaToHydra/cpp/testUsdPointInstancingIsolateSelect.py (hunk 3 above)
Comment (ppt-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/cpp/testUsdPointInstancingIsolateSelect.py]: Good catch, removed.

File: lib/flowViewport/sceneIndex/fvpIsolateSelectSceneIndex.cpp (hunk 1 above)
Comment (ppt-adsk) [reply lib/flowViewport/sceneIndex/fvpIsolateSelectSceneIndex.cpp]: No, that was the whole point of factoring it out, done.

**Review by lanierd-adsk — APPROVED** [review]

**Pull request merged**